lua/arrow/buffer_ui.lua

```diff
@@ -63,9 +72,11 @@ end
 
 local function close_preview_windows()
 	for _, buffer in ipairs(preview_buffers) do
-		if vim.api.nvim_win_is_valid(buffer.win) then
-			vim.api.nvim_win_close(buffer.win, true)
-		end
+		vim.schedule(function()
+			if vim.api.nvim_win_is_valid(buffer.win) then
+				vim.api.nvim_win_close(buffer.win, true)
+			end
+		end)
```

Comment:
This can prevent some error? At least makes treesitter-context correctly close windows after arrow window closes.